### ISSUE TITLE
NotificationTopicArn should be parameterized by environment

### ISSUE THREAD

**Issue by francescosolari0** [aws-samples/iam-identity-center-team#152]
I'm creating this issue since the old one regarding this same problem has been closed and I can't re-open it:
Reference: Issue #108

Currently it is not possible to deploy the stack on two different environments because the exported variable has a unique name not parameterized by environment

amplify/backend/custom/sns/sns-cloudformation-template.json
"NotificationTopicArn": {
"Name": "NotificationTopicArn"

After creating the staging environment, if you try to deploy the production environment you get this error

Embedded stack arn:aws:cloudformation:eu-west-1:xxxxxxxx:stack/amplify-teamidcapp-prod-84031-customsns-6QRL9UWQ1MO4/3f292f00-73e6-11ee-80ee-029f6591487d was not successfully created: Export with name NotificationTopicArn is already exported by stack amplify-TEAM-IDC-APP-stag-192729-customsns-1RQ9LAC5A21UI

**Comment by rubenceroni** [comment]
+1 this breaks a lot of stuff

**Comment by vinnymachacek** [comment]
+1 - this is particularly difficult to fix because of the imports that exist in dependent stacks.  Even if you move remove the imports, and pass the values thru the dependency tree in backend-config.json, two deployments would be required to fix this.  Perhaps this is something that should be bundled into the next "breaking" release?

**Comment by randazzop** [comment]
+1

**Comment by vinnymachacek** [comment]
It looks like the problematic cloudformation Export was introduced in the following commit: https://github.com/aws-samples/iam-identity-center-team/pull/54/files#diff-1411e5a1f1a77dfc8635b1af20c128b960a9076b8c24dbe27f9cade4f5936780R34 

In addition to causing issues with deploying more than one amplify branch to the same account, this also creates difficulty with deleting and recreating the stack, as the nested stack deletion will fail as a result of the use of Fn::ImportValue.

The body of work to fix this is something along the lines of:
- Utilize amplify dependencies in `amplify/backend/backend-config.json` to pass the NotificationTopicArn between the custom SNS stack `amplify/backend/custom/sns/sns-cloudformation-template.json` and child stacks
- Remove the use of cloudformation exports from the project entirely
- Additional work to remove hardcoded references to 'TEAM-IDC-APP' and 'main' throughout the code base (10-15 spots) and parameterize by branch
- *Manually* deploy the child stacks (with the now parameterized NotificationTopicArn) to break the dependency of those stacks on the cloudformation export/parent stack.

Since this would involve a manual/"breaking change", perhaps we could do a few things to provide a path to fixing this issue in existing deployments:
- Have `deployment/update.sh`  detect if the export exists in the region, and provide use with verbiage/documentation on how to manually remediate
- Provide a script for this remediation, that would manually update the nested stacks impacted by this to remove the dependency on the ImportValue, allowing for the Export to be removed from the parent stack via an amplify deploy.
- Document the breaking change in release notes and put the onus on the customer to fix.

@tawoyinfa is this something we can prioritize fixing as part of the next version release? I am happy to submit a PR for the above *if we can work out a way to get this merged into main in an agreeable fashion*

**Comment by tawoyinfa** [comment]
@vinnymachacek  thanks for volunteering to submit a PR for this. TEAM wasnt designed for mutiple implementations in a single account (due to it's dependence on a single instance of IDC). At this stage, I would optimise for stability as opposed to big step changes. I would be happy to review and merge the PR  if it would not massively impact existing deployments.

**Comment by randazzop** [comment]
Hi @tawoyinfa it is correct that there is only one IDC however in my experience it is better to have two environments, one test environment where to verify new versions (accessible only to admins) and one production environment dedicated to users for daily operations. TEAM is a beautiful tool in continuous development and having two environments is desirable in order not to block users' operations in case of deployment issues or blocking bugs (rare but can happen).

I'am available to test @vinnymachacek 's solution if accepted, thank you both!

**Comment by vinnymachacek** [comment]
#### Proposed Release Notes for this change
Upgrade Instructions v X.Y.Z

This version changes how the SNS Notification Arn is passed in order to unblock Issue #152. Deploying this version requires a few manual steps:

Three cloudformation stacks will need to be manually updated *before* running the update.sh script to trigger an amplify deployment. 

You can list these stacks using the AWS CLI like shown below:


```
aws cloudformation list-imports --export-name NotificationTopicArn
{
    "Imports": [
        "amplify-teamidcapp-main-180021-customstepfunctions-SDFGSFGYHSFTUY",
        "amplify-teamidcapp-main-180021-functionteamRouter-SFGHSFGSDFG",
        "amplify-teamidcapp-main-180021-functionteamNotifications-SDFGSDFGSDFH"
    ]
}
```

You can find the corresponding cloudformation template json in the ./amplify/backend/custom and ./amplify/backend/function directory.

When updating the stacks, you'll have to specify a new Parameter `customsnsNotificationTopicArnOutput`. You can obtain the value using the AWS CLI:
```
aws cloudformation list-exports --query 'Exports[?Name==`NotificationTopicArn`]' 
[
    {
        "ExportingStackId": "arn:aws:cloudformation:us-east-2:123412341234:stack/amplify-teamidcapp-main-180021-customsns-T20ZX7BMXSWW/8b79b450-bed3-11ee-b95d-0a1a1cd14f73",
        "Name": "NotificationTopicArn",
        "Value": "arn:aws:sns:us-east-2:123412341234:TeamNotifications-main"
    }
]
```

You should now be able to run `deploy/update.sh` to deploy the application normally
